### PR TITLE
Bs4 faa health coverage page

### PR DESCRIPTION
**Pull request by scaustin34**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: 

# A brief description of the changes

Current behavior:

New behavior:

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/helpers/portal_header_helper.rb
Comment (charlienparker) [inline app/helpers/portal_header_helper.rb:1]: Are these changes intended? I've seen them come up on my PRs after a faulty rebase

File: app/javascript/css/main.scss [hunk 1]
```diff
@@ -514,4 +510,9 @@ div.disabled *:not(:has(*)):not(button, a.button) {
 a.disabled {
   pointer-events: none;
   cursor: not-allowed;
-}
\ No newline at end of file
+}
+
+a.disabled {
```
Comment (charlienparker) [inline app/javascript/css/main.scss:515]: repeat of above

File: app/helpers/portal_header_helper.rb
Comment (ATBull81) [reply app/helpers/portal_header_helper.rb:1]: These have popped up for me as well, should be good to remove I'd think

File: app/views/shared/_is_applying_for_coverage.html.erb [hunk 2]
```diff
@@ -19,13 +19,13 @@
   <div class="col-lg-2 col-md-2 col-sm-2 col-xs-2 form-group form-group-lg no-pd">
     <div tabindex="0" onkeydown="handleRadioKeyDown(event, 'is_applying_coverage_true')" class="radio">
       <%= f.radio_button :is_applying_coverage, true, class: "required floatlabel", id: 'is_applying_coverage_true', checked: first_checked%>
-      <label for="is_applying_coverage_true"><span class="yes_no_pair"><%= l10n("yes") %></span></label>
+      <label for="is_applying_coverage_true"><span class="yes_no_pair">Yes</span></label>
```
Comment (charlienparker) [inline app/views/shared/_is_applying_for_coverage.html.erb:22]: any reason why we removed the l10n here?

File: app/views/shared/_is_applying_for_coverage.html.erb [hunk 3]
```diff
@@ -19,13 +19,13 @@
...
     </div>
   </div>
   <div class="col-lg-2 col-md-2 col-sm-2 col-xs-6 form-group form-group-lg no-pd">
     <div tabindex="0" onkeydown="handleRadioKeyDown(event, 'is_applying_coverage_false')" class="radio">
       <%= f.radio_button :is_applying_coverage, false, class: "required floatlabel", id: 'is_applying_coverage_false', checked: second_checked %>
-      <label for="is_applying_coverage_false"><span class="yes_no_pair"><%= l10n("no") %></span></label>
+      <label for="is_applying_coverage_false"><span class="yes_no_pair">No</span></label>
```
Comment (charlienparker) [inline app/views/shared/_is_applying_for_coverage.html.erb:28]: any reason why we removed the l10n here?

File: components/financial_assistance/app/controllers/financial_assistance/benefits_controller.rb [hunk 4]
```diff
@@ -6,6 +6,9 @@ class BenefitsController < FinancialAssistance::ApplicationController
 
     before_action :find_application_and_applicant
     before_action :set_cache_headers, only: [:index]
+    before_action :enable_bs4_layout, only: [:index, :create, :update] if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
```
Comment (charlienparker) [inline components/financial_assistance/app/controllers/financial_assistance/benefits_controller.rb:9]: I think on my PR I had this setup such that create and update set bs4 based on the params. see `conditionally_enable_bs4_layout` in `incomes_controller`

probably fine to keep this here, I just was not 100% if these actions were ever being called from an unsupported page for bs4

File: components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb [hunk 5]
```diff
@@ -145,6 +145,41 @@
...
   "en.faa.has_dependent_with_coverage" => "Did this person have coverage through a job (for example, a parent’s job) that ended in the last 3 months?*",
   "en.faa.dependent_job_end_on" => "What was the last day this person had coverage through the job?*",
+  "en.faa.health_coverage.title" => "Health Coverage for %{applicant}",
+  "en.faa.health_coverage.coverage_title" => "Health Coverage %{number}",
+  "en.faa.health_coverage.desc" => "Tell us about other health coverage options for this person. When you're finished, select 'Continue to Next Step'.",
+  "en.faa.health_coverage.coverage_period_question" => "When was the coverage period for marketplace coverage?",
+  "en.faa.health_coverage.remove_income_type" => "Remove Income Type",
+  "en.faa.health_coverage.add_cov_type" => "Add Additional Health Coverage Type",
+  "en.faa.health_coverage.health_cov_type" => "Other Health Coverage Type",
+  "en.faa.health_coverage.edit" => "Edit",
```
Comment (charlienparker) [inline components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb:155]: do edit delete and save need to have their own keys for health coverage? surely they are already defined elsewhere...

File: components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb [hunk 6]
```diff
@@ -145,6 +145,41 @@
...
+  "en.faa.health_coverage.delete" => "Delete",
+  "en.faa.health_coverage.save" => "Save",
+  "en.faa.health_coverage.type_modal_header" => "Health Care Types",
+  "en.faa.health_coverage.esi.employer_name" => "Employer Name",
+  "en.faa.health_coverage.esi.employer_address1" => "Employer Address Line 1",
+  "en.faa.health_coverage.esi.employer_address2" => "Employer Address Line 2",
+  "en.faa.health_coverage.esi.employer_city" => "City",
```
Comment (charlienparker) [inline components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb:162]: do these address strings need to have their own keys for health coverage? surely they are already defined elsewhere...

File: db/seedfiles/translations/en/cca/insured.rb [hunk 7]
```diff
@@ -211,7 +211,7 @@
   :'en.ssn_not_present' => "SSN not present",
   :'en.tax_covering_applicant_not_present' => "Tax covering applicant not present",
   :'en.information_complete' => "Information Complete",
-  :'en.add_person_to_household' => "Add Person to Household",
+  :'en.add_person_to_household' => "Add Member to Household",
```
Comment (charlienparker) [inline db/seedfiles/translations/en/cca/insured.rb:214]: any concern outright changing the legacy string for non-bs4? I know we talked about this before and we think it's probably ok... just want to confirm

Comment (ATBull81) [reply db/seedfiles/translations/en/cca/insured.rb:214]: I think this string was added recently and is only used in the bs4 section of the FAA pages so it should be safe to change. This change is suggested by https://www.pivotaltracker.com/n/projects/2640062/stories/187863301 as well.

File: components/financial_assistance/app/controllers/financial_assistance/benefits_controller.rb (hunk 4 above)
Comment (kristinmerbach) [reply components/financial_assistance/app/controllers/financial_assistance/benefits_controller.rb:9]: These actions aren't called from anywhere else, so safe to just do the simple thing here.

File: app/helpers/portal_header_helper.rb
Comment (kristinmerbach) [reply app/helpers/portal_header_helper.rb:1]: Removed!

File: app/views/shared/_is_applying_for_coverage.html.erb (hunk 2 above)
Comment (kristinmerbach) [reply app/views/shared/_is_applying_for_coverage.html.erb]: no idea, added back in

File: app/views/shared/_is_applying_for_coverage.html.erb (hunk 3 above)
Comment (kristinmerbach) [reply app/views/shared/_is_applying_for_coverage.html.erb]: same as above

File: components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb (hunk 5 above)
Comment (kristinmerbach) [reply components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb:169]: They don't need to, I'm leaving for now

File: components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb (hunk 6 above)
Comment (kristinmerbach) [reply components/financial_assistance/db/seedfiles/translations/en/me/financial_assistance.rb:176]: same as above
